crates/sui-json-rpc/src/event_api.rs

```diff
@@ -58,7 +58,7 @@ impl EventStreamingApiServer for EventStreamingApiImpl {
             event.map(|event| SuiEventEnvelope {
                 timestamp: e.timestamp,
                 tx_digest: e.tx_digest,
-                id: EventID::from((e.seq_num as i64, e.event_num as i64)),
+                id: EventID::from((e.tx_digest.unwrap(), e.event_num as i64)),
```

Comment:
we should make EventEnvelope::tx_digest non-optional, I don't think this is ever None